dask/array/slicing.py

```diff
@@ -561,7 +561,7 @@ def slicing_plan(chunks, index):
         index = np.asanyarray(index)
     cum_chunks = cached_cumsum(chunks)
 
-    cum_chunks = asarray_safe(cum_chunks, like=index)
+    cum_chunks = asarray_safe(cum_chunks, like=index, dtype=index.dtype)
```

Comment:
cum_chunks was int64 before (in numpy 1 too); it now changes to uint64.
This fixes the dtype arithmetics on line 581 below, which used to subtract uint64 - int64, which in numpy 1 returns uint64 but in numpy 2 returns float.